benchmarks/sidekiq_test.rb

```diff
@@ -0,0 +1,134 @@
+# rubocop:disable all
+ENV['RAILS_ENV'] = 'production'
+
+require 'bundler/setup'
+require 'rails/all'
+Bundler.require(*Rails.groups)
+
+module SampleApp
+  class Application < Rails::Application; end
+end
+
+module OverrideConfiguration
+  def paths
+    super.tap { |path| path.add 'config/database', with: 'benchmarks/postgres_database.yml' }
+  end
+end
+
+Rails::Application::Configuration.prepend(OverrideConfiguration)
+
+Rails.application.configure do
+  config.cache_classes = true
+  config.eager_load = true
+  config.active_job.queue_adapter = :sidekiq
+end
+
+ActiveRecord::Base.configurations = Rails.application.config.database_configuration
+Rails.application.initialize!
+
+ActiveRecord::Schema.define do
+  drop_table(:samples) if connection.table_exists?(:samples)
+
+  create_table :samples do |t|
+    t.string :name
+    t.timestamps
+  end
+end
+
+class Sample < ActiveRecord::Base; end
+
+require 'sidekiq/launcher'
+require 'sidekiq/cli'
+require 'concurrent/atomic/atomic_fixnum'
+
+Sidekiq.configure_server do |config|
+  redis_conn = proc { Redis.new(host: 'localhost', port: 6379) }
```

Comment:
Does this need to be configurable? Redis might not be running on the localhost, particularly in a Docker environment.